ServerCore/Pages/Puzzles/Index.cshtml

```diff
@@ -98,9 +98,11 @@
                 @Html.DisplayFor(modelItem => item.MaterialsUrlString)
             </td>
             <td>
-                <a asp-Page="./Edit" asp-route-id="@item.ID">Edit</a> |
-                <a asp-Page="./Details" asp-route-id="@item.ID">Details</a> |
-                <a asp-Page="./Delete" asp-route-id="@item.ID">Delete</a>
+                <a asp-page="./Edit" asp-route-id="@item.ID">Edit</a> |
+                <a asp-page="./Details" asp-route-id="@item.ID">Details</a> |
+                <a asp-page="./Delete" asp-route-id="@item.ID">Delete</a> |
+                <a asp-page="/Responses/Index" asp-route-puzzleid="@item.ID" asp-route-eventid="@item.Event.ID">Responses</a> |
```

Comment:
When you make the Responses pages Event-aware (see link in another comment), you shouldn't need to respecify the eventid here. It should transfer just fine.